### PR TITLE
feat(tokens): add DS tokens

### PR DESCRIPTION
**Pull request by Mnau-mnau**
implements [UXD 633](https://zitenote.atlassian.net/browse/UXD-633) and [UXD 631](https://zitenote.atlassian.net/browse/UXD-631)

tokens formatting open to discussion

### CODE REVIEW

File: src/utils/helpers.ts [hunk 1]
```diff
@@ -167,5 +170,17 @@ export const getNegativeSpace = curry(
     pipe(path(['space', size]), negate, pxToRem)(theme),
 );
 
+// getToken:: Type -> Props -> Definition -> string
+// Type: component - element | button
+//  Props: { variant, color } | { variant: 'text', color: 'danger'}
+//  Definition: state - property - category | 'hoverOutlineColor'
+const capitalize = (string) => string.charAt(0).toUpperCase() + string.slice(1);
+
+export const findToken = curry((type, { variant, color }, definition) => {
```
Comment (ajkl2533) [inline src/utils/helpers.ts:179]: Is it `findToken` or `getToken`?

Comment (Mnau-mnau) [reply src/utils/helpers.ts:179]: facepalm here, I'll rename it

File: src/utils/helpers.ts [hunk 2]
```diff
@@ -167,5 +169,11 @@ export const getNegativeSpace = curry(
     pipe(path(['space', size]), negate, pxToRem)(theme),
 );
 
+// getToken:: String -> Token
+export const capitalize = (string) =>
+  string.charAt(0).toUpperCase() + string.slice(1);
+
+export const getToken = (string) => prop(string, tokens);
```
Comment (ajkl2533) [inline src/utils/helpers.ts:176]:
```suggestion
export const getToken = curry(
  (name: keyof typeof tokens, {theme}: Theme): string => 
    path(['tokens', name])(theme),
  );
```

File: src/components/_internal/BaseButton/BaseStyledButton.tsx [hunk 3]
```diff
@@ -162,38 +182,44 @@ const ButtonText = css<BaseStyledButtonProps>`
...
             ${focusState};
           }
 
           &:not(:disabled):active,
           &:not([href]):not([tabindex]):not(:disabled):active,
           &&&.active {
-            color: ${getButtonColor('activeColor')};
-            text-decoration: none;
+            color: ${getToken(`buttonText${capitalize(color)}TextActiveColor`)};
+            text-decoration: none;xx
```
Comment (ajkl2533) [inline src/components/_internal/BaseButton/BaseStyledButton.tsx:222]: remove `xx`

File: src/theme/theme.ts [hunk 4]
```diff
@@ -11,6 +11,7 @@ import { createDepths } from './depths';
 import { createSpace } from './space';
 import { createRadii } from './radii';
 import { RecursivePartial } from '../types/utils.types';
+import { tokens as themeTokens } from './tokens';
```
Comment (ajkl2533) [inline src/theme/theme.ts:14]:
```suggestion
import { createsTokens } from './tokens';
```

File: src/theme/theme.ts [hunk 5]
```diff
@@ -24,10 +25,12 @@ export const createTheme = (
     depths: depthsOverride = {},
     space: spaceOverride = {},
     radii: radiiOverride = {},
+    tokens: tokensOverride = {},
     ...rest
   } = overrides;
 
   const colors = mergeDeepRight(themeColors, colorsOverride);
+  const tokens = mergeDeepRight(themeTokens, tokensOverride);
```
Comment (ajkl2533) [inline src/theme/theme.ts:33]:
```suggestion
  const tokens = mergeDeepRight(createTokens(colors), tokensOverride);
```

File: src/types/definitions/styled.d.ts [hunk 6]
```diff
@@ -19,5 +20,6 @@ declare module 'styled-components' {
     depths: ReturnType<typeof createDepths>;
     space: ReturnType<typeof createSpace>;
     radii: ReturnType<typeof createRadii>;
+    tokens: typeof tokens;
```
Comment (ajkl2533) [inline src/types/definitions/styled.d.ts:23]:
```suggestion
    tokens: ReturnType<typeof createTokens>;
```

File: src/types/definitions/styled.d.ts [hunk 7]
```diff
@@ -8,6 +8,7 @@ import { createLayout } from '../../theme/layout';
 import { createRadii } from '../../theme/radii';
 import { createSpace } from '../../theme/space';
 import { createTypography } from '../../theme/typography';
+import { tokens } from '../../theme/tokens';
```
Comment (ajkl2533) [inline src/types/definitions/styled.d.ts:11]:
```suggestion
import { createTokens } from '../../theme/tokens';
```

File: src/utils/helpers.ts [hunk 8]
```diff
@@ -167,5 +168,14 @@ export const getNegativeSpace = curry(
     pipe(path(['space', size]), negate, pxToRem)(theme),
 );
 
+// getToken:: String -> Token
+export const capitalize = (string) =>
+  string.charAt(0).toUpperCase() + string.slice(1);
+
+export const getToken = curry(
+  (name: keyof typeof tokens, { theme }: Theme): string =>
```
Comment (ajkl2533) [inline src/utils/helpers.ts:176]:
```suggestion
  (name: keyof ReturnType<typeof createTokens>, { theme }: Theme): string =>
```

**Review by ajkl2533 — APPROVED** [review]
LGTM

**Pull request merged**